airflow/providers/google/cloud/operators/dataplex.py

```diff
@@ -610,3 +612,733 @@ def execute(self, context: Context) -> None:
+
+    def __init__(
+        self,
+        project_id: str,
+        region: str,
+        data_scan_id: str,
+        job_id: str | None = None,
+        api_version: str = "v1",
+        retry: Retry | _MethodDefault = DEFAULT,
+        timeout: float | None = None,
+        metadata: Sequence[tuple[str, str]] = (),
+        gcp_conn_id: str = "google_cloud_default",
+        impersonation_chain: str | Sequence[str] | None = None,
+        fail_on_dq_failure: bool = False,
+        wait_for_results: bool = True,
+        *args,
+        **kwargs,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.project_id = project_id
+        self.region = region
+        self.data_scan_id = data_scan_id
+        self.job_id = job_id
+        self.api_version = api_version
+        self.retry = retry
+        self.timeout = timeout
+        self.metadata = metadata
+        self.gcp_conn_id = gcp_conn_id
+        self.impersonation_chain = impersonation_chain
+        self.fail_on_dq_failure = fail_on_dq_failure
+        self.wait_for_results = wait_for_results
+
+    def execute(self, context: Context) -> dict:
+        hook = DataplexHook(
+            gcp_conn_id=self.gcp_conn_id,
+            api_version=self.api_version,
+            impersonation_chain=self.impersonation_chain,
+        )
+        # fetch the last job
+        if not self.job_id:
+            jobs = hook.list_data_scan_jobs(
+                project_id=self.project_id,
+                region=self.region,
+                data_scan_id=self.data_scan_id,
+                retry=self.retry,
+                timeout=self.timeout,
+                metadata=self.metadata,
+            )
+            job_ids = [DataScanJob.to_dict(job) for job in jobs]
+            if not job_ids:
+                raise AirflowException("There are no jobs, you should create one before.")
+            job_id = job_ids[0]["name"]
+            self.job_id = job_id.split("/")[-1]
+
+        if self.wait_for_results:
+            hook.wait_for_data_scan_job(
+                job_id=self.job_id,
+                data_scan_id=self.data_scan_id,
+                project_id=self.project_id,
+                region=self.region,
+                fail_on_dq_failure=self.fail_on_dq_failure,
+                fail_on_job_failure=False,
+                fail_on_timeout=False,
+                timeout=self.timeout,
+            )
+
+        job = hook.get_data_scan_job(
+            project_id=self.project_id,
+            region=self.region,
+            job_id=self.job_id,
+            data_scan_id=self.data_scan_id,
+            retry=self.retry,
+            timeout=self.timeout,
+            metadata=self.metadata,
+        )
+
+        if self.fail_on_dq_failure:
+            if job.state == DataScanJob.State.SUCCEEDED and not job.data_quality_result.passed:
+                raise AirflowException(f"Data Quality failed: {self.data_scan_id}")
```

Comment:
```suggestion
        
            if job.state == DataScanJob.State.SUCCEEDED:
                if not job.data_quality_result.passed:
                   if  self.fail_on_dq_failure:
                       raise AirflowException(f"Data Quality job execution failed due to failing of its scans: {self.data_scan_id} or something like this, check better explanation")
                else:
                    self.log.info("Data Quality job executed SUCCESSFULLY")
            else:
                self.log.info("Data Quality job execution returned status: %s", job.status)
```
use this check (or similar) for all the places where you are using wait_for_job_state(). It returns status only, without raising errors.

Reply:
changed, 
(here we need not only status but DataScanJob)